analysis_driver/rest_communication.py

```diff
@@ -1,50 +1,68 @@
 from urllib.parse import urljoin
 import requests
-from pprint import pformat
 from analysis_driver.config import default as cfg
 from analysis_driver.app_logging import get_logger
 
 app_logger = get_logger(__name__)
 
 
-def api_url(endpoint):
-    return '{base_url}/{endpoint}/'.format(
+def api_url(endpoint, **query_args):
+    url = '{base_url}/{endpoint}/'.format(
         base_url=cfg.query('rest_api', 'url').rstrip('/'), endpoint=endpoint
     )
+    if query_args:
+        url += '?' + '&'.join(['%s=%s' % (k, v) for k, v in query_args.items()]).replace(' ', '').replace('\'', '"')
+
+    return url
 
 
 def _req(method, url, **kwargs):
     r = requests.request(method, url, **kwargs)
-    if r.status_code != 200:
-        app_logger.debug('%s %s %s %s' % (r.request.method, r.request.path_url, r.status_code, r.reason))
-        json = r.json()
-        if json:
-            app_logger.debug(pformat(json))
+    # e.g: 'POST <url> ({"some": "args"}) -> {"some": "content"}. Status code 201. Reason: CREATED
+    report = '%s %s (%s) -> %s. Status code %s. Reason: %s' % (
+        r.request.method, r.request.path_url, kwargs, r.content.decode('utf-8'), r.status_code, r.reason
+    )
+    if r.status_code in (200, 201):
+        app_logger.debug(report)
+    else:
+        app_logger.error(report)
     return r
 
+
+def depaginate_documents(endpoint, **queries):
```

Comment:
Looks like they can be merged. You already can specify `page=` in the keyword args and it will be passed to Eve (we just haven't been using it). A boolean `depaginate` argument might be cleaner than `page=-1`